sycl/source/detail/memory_manager.cpp

```diff
@@ -218,26 +229,34 @@ void copyD2H(SYCLMemObjT *SYCLMemObj, cl_mem SrcMem, QueueImplPtr SrcQueue,
   SrcSize[0] *= SrcElemSize;
   DstSize[0] *= DstElemSize;
 
-  cl_command_queue CLQueue = UseExclusiveQueue
-                                 ? SrcQueue->getExclusiveQueueHandleRef()
-                                 : SrcQueue->getHandleRef();
-
   if (1 == DimDst && 1 == DimSrc) {
     CHECK_OCL_CODE(clEnqueueReadBuffer(
-        CLQueue, SrcMem,
+        SrcQueue->getHandleRef(), SrcMem,
         /*blocking_read=*/CL_FALSE, DstOffset[0], DstAccessRange[0],
         DstMem + DstOffset[0], DepEvents.size(), &DepEvents[0], &OutEvent));
   } else {
-    size_t BufferRowPitch = (1 == DimSrc) ? 0 : SrcSize[0];
-    size_t BufferSlicePitch = (3 == DimSrc) ? SrcSize[0] * SrcSize[1] : 0;
-
-    size_t HostRowPitch = (1 == DimDst) ? 0 : DstSize[0];
-    size_t HostSlicePitch = (3 == DimDst) ? DstSize[0] * DstSize[1] : 0;
-    CHECK_OCL_CODE(clEnqueueReadBufferRect(
-        CLQueue, SrcMem,
-        /*blocking_read=*/CL_FALSE, &SrcOffset[0], &DstOffset[0],
-        &SrcAccessRange[0], BufferRowPitch, BufferSlicePitch, HostRowPitch,
-        HostSlicePitch, DstMem, DepEvents.size(), &DepEvents[0], &OutEvent));
+    // test if the whole buffer is copied
+    if( SrcSize[0]   == SrcAccessRange[0] &&  SrcSize[1]   == SrcAccessRange[1]  &&  SrcSize[2]   == SrcAccessRange[2] &&
+	SrcSize[0]   == DstSize[0]        &&  SrcSize[1]   == DstSize[1]         &&  SrcSize[2]   == DstSize[2]        &&
+	SrcSize[0]   == DstAccessRange[0] &&  SrcSize[1]   == DstAccessRange[1]  &&  SrcSize[2]   == DstAccessRange[2] &&
+	SrcOffset[0] == DstOffset[0]      &&  SrcOffset[1] == DstOffset[1]       &&  SrcOffset[2] == DstOffset[2]      &&
+	SrcOffset[0] == 0                 &&  SrcOffset[1] == 0                  &&  SrcOffset[2] == 0 ) {
+      CHECK_OCL_CODE(clEnqueueReadBuffer(
+        SrcQueue->getHandleRef(), SrcMem,
```

Comment:
Same request as prior comment

Reply:
And keep the indentation similar to the way it was on the old clEnqueueReadBuffer please